meta/Decays.m

```diff
@@ -584,22 +584,29 @@ would violate Ward identity (as at the level of dim 4
            concreteFinalStates = Join @@ (GetParticleCombinationsOfType[#, allowedFinalStateParticles, isPossibleDecay]& /@ genericFinalStates);
            concreteFinalStates = OrderFinalState[particle, #] & /@ concreteFinalStates;
 
-           Print[""];
            Print["Creating amplitudes for ", particle, " decays..."];
-           (*ParallelEvaluate[Get["/home/wojciech/HEP-software/mathematica/SARAH-4.14.3/SARAH.m"]]
-           ParallelEvaluate[Start["MSSM"]];*)
-           decays =
-              Map[
-                 (
-                    (* @todo StringPadRigh was introduced only in 10.1 *)
-                    WriteString["stdout", StringPadRight["   - Creating amplitude for " <> ToString@particle <> " -> " <> ToString@#, 64, "."]];
-                    temp = FSParticleDecay[particle, #, GetContributingGraphsForDecay[particle, #]];
-                    Print[" Done."];
-                    temp
-                 )&,
-                 concreteFinalStates(*,
-                 DistributedContexts -> All*)
+
+           If[FlexibleSUSY`FSEnableParallelism,
+              decays =
+                 ParallelMap[
+                    FSParticleDecay[particle, #, GetContributingGraphsForDecay[particle, #]]&,
+                    concreteFinalStates,
+                    DistributedContexts -> All,
```

Comment:
I am not sure, how this will interfere with ``NPointFunctions`` code. There is an opinion, that this will force all further slave kernels to have these contexts, which is not good for loading ``FeynArts`` and ``FormCalc``. Also, there is a funny point: ``DistributedContexts`` was introduced in ``Mathematica 8.0``.

Reply:
Not I am sure that this parallelization **kills** ``NPointFunctions`` modules.

Reply:
Does your way of parallelizing consume extra license?

Reply:
No, your way of parallelizing pollutes all slave kernels with definitions, which are incompatible with ``FeynArts`` and ``FormCalc``.

Can you please try to use ``Parallel`Developer`ClearSlaves[]`` in the end of your code and also close all opened slaves?

Reply:
I have found this function today.

Reply:
> No, your way of parallelizing pollutes all slave kernels with definitions, which are incompatible with `FeynArts` and `FormCalc`.

I understand that. But what I was asking is if your parallelization needs an extra license.

Reply:
No, it works, because I have 2 Master and 8 slave kernels. And it seems that ``Parallel`Developer`ClearSlaves[]`` does not clean definitions for slaves for me.

Reply:
I have discovered the following functions as well:
```mathematica
Needs["Parallel`Developer`"];
Parallel`Developer`ClearDistributedDefinitions[];
Parallel`Developer`ClearKernels[];
```